sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala

```diff
@@ -1089,7 +1089,8 @@ class CodegenContext extends Logging {
           // Generate the code for this expression tree and wrap it in a function.
           val fnName = freshName("subExpr")
           val inputVars = inputVarsForAllFuncs(i)
-          val argList = inputVars.map(v => s"${v.javaType.getName} ${v.variableName}")
+          val argList =
+            inputVars.map(v => s"${CodeGenerator.typeName(v.javaType)} ${v.variableName}")
```

Comment:
Oh, is this for only 3.0/3.1/3.2?

Reply:
Yea, in 2.4 we have not split it as function yet.

Reply:
Ah, nice catch! Users can see this compilation error? If yes, it would be nice to add end-2-end tests, too, I think.

Reply:
Yea, there will be a error log. This is hard to come out a e2e test, honestly. Although we add subexpr elimination to Project recently, so it is relatively easy to add a e2e test from a Project, but in branch-3.0 or branch-3.1, only hash aggregation uses subexpr elimination.

This unit test is pretty close to e2e test case. It actually writes the java code and compile it.

Reply:
Good catch